syntax = "proto3";

package fbsrankings.messages.event;

import "fbsrankings/messages/options/options.proto";

message RankingValue {
    string id = 1;
    int32 order = 2;
    int32 rank = 3;
    float value = 4;
}

message TeamRankingCalculatedEvent {
    string event_id = 1;
    string ranking_id = 2;
    string name = 3;
    string season_id = 4;
    optional int32 week = 5;
    repeated RankingValue values = 6;

    option (fbsrankings.messages.options.topic) = "fbsrankings.event.team_ranking_calculated";
}

message GameRankingCalculatedEvent {
    string event_id = 1;
    string ranking_id = 2;
    string name = 3;
    string season_id = 4;
    optional int32 week = 5;
    repeated RankingValue values = 6;

    option (fbsrankings.messages.options.topic) = "fbsrankings.event.game_ranking_calculated";
}
